spec/facility_spec.rb

```diff
@@ -1,26 +1,136 @@
 require 'spec_helper'
+require './lib/facility'
+require './lib/vehicle'
+require './lib/registrant'
+require 'date'
 
 RSpec.describe Facility do
   before(:each) do
-    @facility = Facility.new({name: 'DMV Tremont Branch', address: '2855 Tremont Place Suite 118 Denver CO 80205', phone: '(720) 865-4600'})
-  end
+  @facility_1 = Facility.new({name: 'DMV Tremont Branch', address: '2855 Tremont Place Suite 118 Denver CO 80205', phone: '(720) 865-4600'})
+  
+  @facility_2 = Facility.new({name: 'DMV Northeast Branch', address: '4685 Peoria Street Suite 101 Denver CO 80239', phone: '(720) 865-4600'})
+
+  @cruz = Vehicle.new({vin: '123456789abcdefgh', year: 2012, make: 'Chevrolet', model: 'Cruz', engine: :ice, plate_type: :regular})
+
+  @bolt = Vehicle.new({vin: '987654321abcdefgh', year: 2019, make: 'Chevrolet', model: 'Bolt', engine: :ev})
+
+  @camaro = Vehicle.new({vin: '1a2b3c4d5e6f', year: 1969, make: 'Chevrolet', model: 'Camaro', engine: :ice})
+end
+
+
   describe '#initialize' do
-    it 'can initialize' do
-      expect(@facility).to be_an_instance_of(Facility)
-      expect(@facility.name).to eq('DMV Tremont Branch')
-      expect(@facility.address).to eq('2855 Tremont Place Suite 118 Denver CO 80205')
-      expect(@facility.phone).to eq('(720) 865-4600')
-      expect(@facility.services).to eq([])
+    it 'initializes with correct attributes' do
+      expect(@facility_1.name).to eq('DMV Tremont Branch')
+      expect(@facility_1.address).to eq('2855 Tremont Place Suite 118 Denver CO 80205')
+      expect(@facility_1.phone).to eq('(720) 865-4600')
+      expect(@facility_1.services).to eq([])
+      expect(@facility_1.collected_fees).to eq(0)
     end
   end
 
-  describe '#add service' do
-    it 'can add available services' do
-      expect(@facility.services).to eq([])
-      @facility.add_service('New Drivers License')
-      @facility.add_service('Renew Drivers License')
-      @facility.add_service('Vehicle Registration')
-      expect(@facility.services).to eq(['New Drivers License', 'Renew Drivers License', 'Vehicle Registration'])
+  describe '#add_service' do
+    it 'adds a service to the facility' do
+      @facility_1.add_service('Vehicle Registration')
+      expect(@facility_1.services).to include('Vehicle Registration')
     end
   end
+
+  describe '#register_vehicle' do
+      it 'registers a vehicle with plate_type :regular' do 
+        @facility_1.add_service('Vehicle Registration')
+        @facility_1.register_vehicle(@cruz)
+        expect(@cruz.registration_date).to eq(nil)
+        expect(@cruz.plate_type).to eq(:regular)
+        expect(@facility_1.registered_vehicles).to include(@cruz)
+        expect(@facility_1.collected_fees).to eq(100)
+      end
+
+      it 'registers plate_type :antique' do 
+        @facility_1.add_service('Vehicle Registration')
+        @facility_1.register_vehicle(@camaro)
+        expect(@camaro.registration_date).to eq(nil)
```

Comment:
We actually want this registration_date to NOT be nil once the vehicle is registered.